app/Http/Controllers/AssetModelsController.php

```diff
@@ -90,6 +91,11 @@ public function store(ImageUploadRequest $request)
 
         // Was it created?
         if ($model->save()) {
+            if ($request->filled('eol')) {
+                $newEol = $model->eol; 
+                $model->assets()->whereNotNull('purchase_date')->where('eol_explicit', false)
+                    ->update(['asset_eol_date' => DB::raw('DATE_ADD(purchase_date, INTERVAL :newEol MONTH)', ['newEol' => $newEol])]); 
```

Comment:
This is for performance reasons, foreaching was just too slow, the `DB::raw` is because there's not a way to access another column from within an `update` in eloquent.